lib/src/test/java/io/ably/lib/test/realtime/RealtimePresenceHistoryTest.java

```diff
@@ -1292,4 +1292,124 @@ public void run() {
 				rxAbly.close();
 		}
 	}
+
+	/**
+	 * Connect twice to the service, each using the default (binary) protocol.
+	 * Publish messages on one connection to a given channel; while in progress,
+	 * attach the second connection to the same channel and verify a message
+	 * history up to the point of attachment can be obtained.
+	 */
+	@Test
+	public void presencehistory_until_attach() {
+		AblyRealtime txAbly = null, rxAbly = null;
+		try {
```

Comment:
As above, I think we should make this test more deterministic and less effected by network conditions